src/colmap/feature/lightglue.cc

```diff
@@ -0,0 +1,262 @@
+    matches->reserve(num_keypoints1);
+    for (int i = 0; i < num_keypoints1; ++i) {
+      const int64_t j = matches_ptr[i];
+      if (j >= 0) {
+        FeatureMatch match;
+        match.point2D_idx1 = i;
+        match.point2D_idx2 = j;
+        THROW_CHECK_LT(match.point2D_idx2, num_keypoints2);
+        matches->push_back(match);
+      }
+    }
+  }
+
+  void MatchGuided(double max_error,
+                   const Image& image1,
+                   const Image& image2,
+                   TwoViewGeometry* two_view_geometry) override {
+    THROW_CHECK_GE(max_error, 0);
+    Match(image1, image2, &two_view_geometry->inlier_matches);
+  }
+
+ private:
+  torch::Dict<std::string, torch::Tensor> FeaturesFromImage(
+      const Image& image) {
+    THROW_CHECK_NE(image.image_id, kInvalidImageId);
+    THROW_CHECK_NOTNULL(image.descriptors);
+    THROW_CHECK_NOTNULL(image.keypoints);
+    THROW_CHECK_EQ(image.descriptors->rows(), image.keypoints->size());
+
+    bool requires_scale = false;
+    bool requires_orientation = false;
+    int descriptor_dim_size = 0;
+    float descriptor_normalization = 1.0f;
+    torch::TensorOptions descriptor_options;
+    switch (options_.type) {
+      case FeatureMatcherType::LIGHTGLUE_SIFT:
+        requires_scale = true;
+        requires_orientation = true;
+        descriptor_dim_size = sizeof(uint8_t);
+        descriptor_normalization = 1.0f / 512.0f;
+        descriptor_options = torch::TensorOptions().dtype(torch::kUInt8);
+        break;
+      case FeatureMatcherType::LIGHTGLUE_ALIKED:
+        descriptor_dim_size = sizeof(float);
+        descriptor_options = torch::TensorOptions().dtype(torch::kFloat32);
+        break;
+      default:
+        throw std::runtime_error("Unknown descriptor data type provided");
+    }
+
+    const int num_keypoints = image.keypoints->size();
+    THROW_CHECK_EQ(image.descriptors->cols() % descriptor_dim_size, 0);
+    const int descriptor_dim = image.descriptors->cols() / descriptor_dim_size;
+
+    // Convert to contiguous array so we can use torch::from_blob below.
+    std::vector<float> keypoints_array(num_keypoints * 2);
+    for (int i = 0; i < num_keypoints; ++i) {
+      const FeatureKeypoint& keypoint = (*image.keypoints)[i];
+      keypoints_array[2 * i + 0] = keypoint.x;
+      keypoints_array[2 * i + 1] = keypoint.y;
+    }
+
+    const auto kFloat32Options = torch::TensorOptions().dtype(torch::kFloat32);
+
+    torch::Tensor torch_keypoints =
+        torch::from_blob(
+            keypoints_array.data(), {num_keypoints, 2}, kFloat32Options)
+            .clone();
+    // Normalize to [-1, 1].
+    torch_keypoints.index({torch::indexing::Ellipsis, 0}) =
+        (2.0f * torch_keypoints.index({torch::indexing::Ellipsis, 0}) /
+         image.width) -
+        1.0f;
+    torch_keypoints.index({torch::indexing::Ellipsis, 1}) =
+        (2.0f * torch_keypoints.index({torch::indexing::Ellipsis, 1}) /
+         image.height) -
+        1.0f;
```

Comment:
Isn't this already done in `LightGlue::forward` by passing the image size as input?